python/cuml/neighbors/kneighbors_classifier.pyx

```diff
@@ -212,24 +206,24 @@ class KNeighborsClassifier(NearestNeighbors):
             When set to True, the fit method will automatically
             convert the inputs to np.float32.
         """
+
+        out_type = self._get_output_type(X)
```

Comment:
The output_type should be getting passed into the constructor so that user's can control if they always want the same output type or if the output should match the input. The constructor in `cuml.Base` also accepts an `output_type` for this reason.